src/main/kotlin/org/dashevo/dapiclient/DapiClient.kt

```diff
@@ -1,219 +1,398 @@
+            this(SingleMasternode(masternodeAddress), diffMasternodeEachCall)
+
+    /* Platform gRPC methods */
 
     /**
-     * Create a DAP
-     * @param dapSchema Schema for the DAP being Created
-     * @param userRegTxId hash of Blockchain User registration transaction (SubTx)
-     * @param cb callback, instance of {@link DapCallback}
+     * Send State Transition to machine
+     *
+     * @param stateTransition
      */
-    open fun registerDap(dapSchema: Map<String, Any>, userRegTxId: Sha256Hash, hashPrevSubTx: Sha256Hash,
-                         privKey: ECKey, cb: DapiRequestCallback<String>) {
+    fun applyStateTransition(stateTransition: StateTransition) {
+        val applyStateTransitionRequest = PlatformOuterClass.ApplyStateTransitionRequest.newBuilder()
+                .setStateTransition(ByteString.copyFrom(stateTransition.serialize()))
+                .build()
 
-        sendStateTransition(ContractSTPacket("", Contract("", mutableMapOf())), userRegTxId, hashPrevSubTx, privKey, cb)
+        val service = getPlatformService()
+
+        try {
+            service.applyStateTransition(applyStateTransitionRequest)
+        } catch (e: StatusRuntimeException) {
+            logger.log(Level.WARNING, "RPC failed: {0}", e.status)
+            throw e
+        }
     }
 
     /**
-     * Send DAP Object.
-     * @param dapObject: Object to be registered.
-     * @param dapId: DAP ID that the object belongs to.
-     * @param userRegTxId: Transaction ID of User creating the object.
-     * @param hashPrevSubTx: Transaction ID of User's last State Transition or [userRegTxId] if it's the first one.
-     * @param privKey: Private Key of User's pubKey.
-     * @param cb: Callback.
+     * Fetch the identity by id
+     * @param {string} id
+     * @returns {Promise<!Buffer|null>}
      */
-    fun sendDapObject(dapObject: MutableMap<String, Any>, dapId: String, userRegTxId: Sha256Hash, hashPrevSubTx: Sha256Hash,
-                      privKey: ECKey, cb: DapiRequestCallback<String>) {
+    fun getIdentity(id: String): ByteString? {
+        val getIdentityRequest = PlatformOuterClass.GetIdentityRequest.newBuilder()
+                .setId(id)
+                .build()
+
+        val service = getPlatformService()
 
-        val document = Document(dapObject)
-        val stPacket = DocumentsSTPacket(dapId, listOf(document))
-        sendStateTransition(stPacket, userRegTxId, hashPrevSubTx, privKey, cb)
+        val getIdentityResponse: PlatformOuterClass.GetIdentityResponse
+        try {
+            getIdentityResponse = service.getIdentity(getIdentityRequest);
+
+            val serializedIdentityBinaryArray = getIdentityResponse.identity
+
+            return if(!serializedIdentityBinaryArray.isEmpty)
+                serializedIdentityBinaryArray
+            else null
+        } catch (e: StatusRuntimeException) {
+            if (e.status.code == Status.NOT_FOUND.code) {
+                return null;
```

Comment:
Semicolons are not needed in Kotlin. You can remove them by using the `Analyze -> Code Cleanup` menu option.

Reply:
Thanks, sometimes is hard to switch back and forth between java and kotlin.

Reply:
Cleaned up here: f373074